components/PlantDetails.js

```diff
@@ -0,0 +1,84 @@
+import Image from "next/image";
+import Link from "next/link";
+import styled from "styled-components";
+import { useRouter } from "next/router";
+import useLocalStorageState from "use-local-storage-state";
+
+const lightNeedIcon = {
+  "Full Sun": "☀️☀️☀️☀️",
+  "Partial Shade": "☀️☀️☀️",
+  Shade: "☀️☀️",
+  "Full Shade": "☀️",
+};
+
+const waterNeedIcon = {
+  Low: "💧",
+  Medium: "💧💧",
+  High: "💧💧💧",
+};
+
+const seasonIcons = {
+  Spring: "🌱",
+  Summer: "🐝",
+  Fall: "🍂",
+  Winter: "❄️",
+};
+
+export default function PlantDetails() {
+  const router = useRouter();
+  const { id } = router.query;
+
+  const [favourites] = useLocalStorageState("favourites", {
```

Comment:
This can lead to bugs. For example if a user first visits a detailspage, the local storage will be an empty array even if he visits the index.js afterwards.
Since you read the data from local storage in _app.js you should better pass it as props.